Makefile

```diff
@@ -0,0 +1,55 @@
+GOLANGCI_VERSION=v1.28.3
+
+.PHONY: setup
+setup: ## Setup any developer dependencies
+	@echo "==>"
+	@echo "==> Installing dependencies..."
+	@curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s $(GOLANGCI_VERSION)
+
+.PHONY: fmt
```

Comment:
as mentioned in https://github.com/mongodb-forks/digest/pull/2, I think makefiles like these are superfluous and I'd rather not have one unless we are really using them in such a way that the commands are too complex or cumbersome to write/remember yourself

Reply:
comment was resolved here but I still don't think a Makefile is very useful for a project such as this along with pre-commit hook since I don't think most cloning will actually add the hook in. Can we keep this removed for now. I'd rather we just make an evg project that runs what's in here in a an .evg.yml task or two or three

Reply:
The benefit of having a makefile is to provide a well known (and easy) developer experience.

When I see a Makefile, I instinctively know where to look to quickly grasp the common developer flows for that project.

Also, it helps with things like git hooks, as any committers can easily run the same make command that would run in CI.

Reply:
I think the flows are inherit with golang though. Most editors/IDEs do your linting and gofmting for you and really as this is a library, there's nothing to make. testing is also inherit with golang doing just `go test ./...`

Reply:
most != all :) I'd rather have a makefile that standardizes the 'source of truth' rather than assuming that every contributor will have the same editor configuration (or having to explain in the documentation what they should set up).  I like Makefiles because they express intent and because they are easily consumable with zero effort from contributors.